### PR TITLE
RSDK-1740, RSDK-1745 add request logging / idiomatic responses

### PR DESCRIPTION
**Pull request by maximpertsov**
JIRA: https://viam.atlassian.net/browse/RSDK-1740, https://viam.atlassian.net/browse/RSDK-1745

* Adds client config option to log grpc requests
* Updates some wrapper return signatures to return primitive values instead of protobuf responses

### CODE REVIEW

File: src/components/base/Client.ts [hunk 1]
```diff
@@ -26,11 +29,12 @@ export class BaseClient implements Base {
     request.setDistanceMm(distanceMm)
     request.setExtra(googleProtobufStructPb.Struct.fromJavaScript(extra))
 
-    const response = await promisify<baseApi.MoveStraightRequest, baseApi.MoveStraightResponse>(
+    this.options.requestLogger?.(request)
```
Comment (Kschappacher) [inline src/components/base/Client.ts:32]: Im curious, what happens when requestLogger is not set and this line is called. will it break? or just not run?

Comment (maximpertsov) [reply src/components/base/Client.ts:32]: the `?.()` syntax makes it so we only execute the call if requestLogger is set

Comment (maximpertsov) [reply src/components/base/Client.ts:32]: but testing on controls page now to verify

Comment (Kschappacher) [reply src/components/base/Client.ts:32]: follow up question, has this been tested in local? and do we need any .bind to make this work?

**Review by Kschappacher — APPROVED** [review]
LGTM

File: src/components/base/Client.ts (hunk 1 above)
Comment (maximpertsov) [reply src/components/base/Client.ts:32]: looks good in local, no need to bind here

**Pull request merged**